dbt/include/spark/macros/materializations/snapshot.sql

```diff
@@ -15,7 +15,12 @@
 {% macro spark__snapshot_merge_sql(target, source, insert_cols) -%}
 
     merge into {{ target }} as DBT_INTERNAL_DEST
-    using {{ source }} as DBT_INTERNAL_SOURCE
+    {% if target.is_iceberg %}
+      {# create view only supports a name (no catalog, or schema) #}
+      using {{ source.identifier }} as DBT_INTERNAL_SOURCE
```

Comment:
Along the same lines as my comment below, can we define `source.schema=None`, and then `source.__str__() == source.identifier` for iceberg?

Reply:
See above.